disperser/batcher/grpc/dispatcher.go

```diff
@@ -153,15 +155,18 @@ func getBlobMessage(blob *core.BlobMessage) (*node.Blob, error) {
 		lengthProofData.YA1 = blob.BlobHeader.LengthProof.Y.A1.Marshal()
 	}
 
-	quorumHeaders := make([]*node.BlobQuorumInfo, len(blob.BlobHeader.QuorumInfos))
+	quorumHeaders := make([]*node.BlobQuorumInfo, 0)
 
-	for i, header := range blob.BlobHeader.QuorumInfos {
-		quorumHeaders[i] = &node.BlobQuorumInfo{
-			QuorumId:           uint32(header.QuorumID),
-			AdversaryThreshold: uint32(header.AdversaryThreshold),
-			ChunkLength:        uint32(header.ChunkLength),
-			QuorumThreshold:    uint32(header.QuorumThreshold),
-			Ratelimit:          header.QuorumRate,
+	for _, header := range blob.BlobHeader.QuorumInfos {
+		if _, ok := blob.Bundles[header.QuorumID]; ok {
```

Comment:
Shoudn't this always be ok?

Reply:
The way it's implemented today, `blob.BlobHeader` is the same for all blobs sent to all operators. So, when an operator is only registered in quorum 1 and blob is dispersed to quorums 0 and 1, `blob.BlobHeader.QuorumInfos` will contain both quorums. 
However, `blob.Bundles` is operator specific, so for operator only in quorum 1, `blob.Bundles[0]` does not exist.

Reply:
Wait, the DA nodes should be verifying the merkle root in the batch header, right? If they aren't, it would be a security hole. 

In order for them to verify the root, they need all the leaves, which means all of the blobHeaders.